bulk-load/src/main/java/no/sikt/nva/data/report/api/etl/transformer/GenerateKeyBatchesHandler.java

```diff
@@ -0,0 +1,150 @@
+        writeObject(toKeyString(keys));
+        var lastEvaluatedKey = getLastEvaluatedKey(keys);
+        var eventsResponse = sendEvent(constructRequestEntry(lastEvaluatedKey, context, location));
+        logger.info(eventsResponse.toString());
+        return null;
+    }
+
+    private String getLocation(KeyBatchRequestEvent event) {
+        return isNotEmptyEvent(event) ? event.getLocation() : DEFAULT_LOCATION;
+    }
+
+    private static boolean isNotEmptyEvent(KeyBatchRequestEvent event) {
+        return nonNull(event) && nonNull(event.getLocation());
+    }
+
+    private static PutEventsRequestEntry constructRequestEntry(String lastEvaluatedKey, Context context,
+                                                               String location) {
+        return PutEventsRequestEntry.builder()
+                   .eventBusName(EVENT_BUS)
+                   .detail(new KeyBatchRequestEvent(lastEvaluatedKey, TOPIC, location).toJsonString())
+                   .detailType(MANDATORY_UNUSED_SUBTOPIC)
+                   // TODO: replace Object.class with actual class name
+                   .source(Object.class.getName())
+                   .resources(context.getInvokedFunctionArn())
+                   .time(Instant.now())
+                   .build();
+    }
+
+    private static String getStartMarker(KeyBatchRequestEvent input) {
+        return notEmptyEvent(input) ? input.getStartMarker() : DEFAULT_START_MARKER;
+    }
+
+    private static boolean notEmptyEvent(KeyBatchRequestEvent event) {
+        return nonNull(event) && nonNull(event.getStartMarker());
+    }
+
+    private static ListObjectsV2Request createRequest(String startMarker, String location) {
+        return ListObjectsV2Request.builder()
+                   .bucket(INPUT_BUCKET)
+                   .prefix(location + DELIMITER)
+                   .delimiter(DELIMITER)
+                   .startAfter(startMarker)
+                   .maxKeys(MAX_KEYS)
+                   .build();
+    }
+
+    private static String toKeyString(List<String> values) {
+        return values.stream().collect(Collectors.joining(System.lineSeparator()));
+    }
+
+    private static List<String> getKeys(ListObjectsV2Response response) {
+        return response.contents().stream().map(S3Object::key).toList();
+    }
+
+    private static String getLastEvaluatedKey(List<String> keys) {
+        return keys.get(keys.size() - 1);
+    }
+
+    @JacocoGenerated
+    private static EventBridgeClient defaultEventBridgeClient() {
+        return EventBridgeClient.builder().httpClientBuilder(UrlConnectionHttpClient.builder()).build();
+    }
+
+    private PutEventsResponse sendEvent(PutEventsRequestEntry event) {
+        return eventBridgeClient.putEvents(PutEventsRequest.builder().entries(event).build());
+    }
+
+    private void writeObject(String object) {
+        var request = PutObjectRequest.builder().bucket(OUTPUT_BUCKET).key(randomUUID().toString()).build();
```

Comment:
Just a thought: Should we name the output object key something meaningful + `randomUUID()`?

Reply:
In essence, no. These should be viewed as ephemeral, non-human-readable. They are in practice commands for the consumer. The metadata you are seeking is the bucket name.